pkg/vsphere/datastore/datastore.go

```diff
@@ -268,54 +252,56 @@ func (d *Helper) IsVSAN(ctx context.Context) bool {
 // result so the URI doesn't need to be recomputed for every datastore
 // operation.
 func (d *Helper) mkRootDir(ctx context.Context, rootdir string) error {
+	if rootdir == "" {
+		return fmt.Errorf("root directory is empty")
+	}
+
+	if path.IsAbs(rootdir) {
+		return fmt.Errorf("root directory (%s) must not be an absolute path", rootdir)
+	}
 
 	// Handle vsan
 	// Vsan will complain if the root dir exists.  Just call it directly and
 	// swallow the error if it's already there.
 	if d.IsVSAN(ctx) {
+		comps := strings.Split(rootdir, "/")
+
 		nm := object.NewDatastoreNamespaceManager(d.s.Vim25())
 
 		// This returns the vmfs path (including the datastore and directory
 		// UUIDs).  Use the directory UUID in future operations because it is
 		// the stable path which we can use regardless of vsan state.
-		uuid, err := nm.CreateDirectory(ctx, d.ds, rootdir, "")
+		uuid, err := nm.CreateDirectory(ctx, d.ds, comps[0], "")
 		if err != nil {
 			if !soap.IsSoapFault(err) {
 				return err
 			}
 
 			soapFault := soap.ToSoapFault(err)
-			_, ok := soapFault.VimFault().(types.FileAlreadyExists)
-			if ok {
+			if _, ok := soapFault.VimFault().(types.FileAlreadyExists); ok {
```

Comment:
if we're restructuring, change this test to `!ok` so that we can directly return and drop the else clause.